src/encryption.rs

```diff
@@ -0,0 +1,123 @@
+        .map_err(|e| PvwError::InvalidParameters(format!("Failed to sample noise: {}", e)))?;
+
+    let e2 = sample_vec_cbd(k, pk.params.variance, rng)
+        .map_err(|e| PvwError::InvalidParameters(format!("Failed to sample noise: {}", e)))?;
+
+    // TODO: Ask if this should be sk params
+    let r = sample_vec_cbd(k, pk.params.variance, rng)
+        .map_err(|e| PvwError::InvalidParameters(format!("Failed to sample r: {}", e)))?;
+
+    let a = pk.crs.clone();
+    let b = pk.matrix.clone();
+
+    // Compute c1 = A * r^T + e1^T mod q
+    // A is k×k matrix, r is k×1 vector, result is k×1 vector
+    let mut c1 = Vec::with_capacity(k);
+    let mut c2 = Vec::with_capacity(k);
+
+    for i in 0..k {
+        // Compute row i of A * r: sum over j of A[i][j] * r[j]
+        let mut row_result_a = Poly::zero(ctx, Representation::Ntt);
+        // Compute row i of B * r: sum over j of B[i][j] * r[j]
+        let mut row_result_b = Poly::zero(ctx, Representation::Ntt);
+
+        // Create polynomial for m[i]
+        let m_poly = Poly::from_coefficients(&[m[i]], ctx).map_err(|e| {
+            PvwError::InvalidParameters(format!("Failed to create m polynomial: {}", e))
+        })?;
+
+        // Create polynomial for e1[i]
+        let mut e1_i_poly = Poly::from_coefficients(&[e1[i].clone()], ctx).map_err(|e| {
+            PvwError::InvalidParameters(format!("Failed to create e1 polynomial: {}", e))
+        })?;
+        e1_i_poly.change_representation(Representation::Ntt);
+
+        // Create polynomial for e2[i]
+        let mut e2_i_poly = Poly::from_coefficients(&[e2[i].clone()], ctx).map_err(|e| {
+            PvwError::InvalidParameters(format!("Failed to create e2 polynomial: {}", e))
+        })?;
+        e2_i_poly.change_representation(Representation::Ntt);
+
+        for j in 0..k {
+            // Create polynomial for r[j]
+            let mut r_j_poly = Poly::from_coefficients(&[r[j].clone()], ctx).map_err(|e| {
+                PvwError::InvalidParameters(format!("Failed to create r[j] polynomial: {}", e))
+            })?;
+            r_j_poly.change_representation(Representation::Ntt);
+
+            // A * r
+            let a_ij = a.get(i, j).ok_or_else(|| {
+                PvwError::InvalidParameters(format!("CRS matrix access failed at ({}, {})", i, j))
+            })?;
+            let product_a = a_ij * &r_j_poly;
+            row_result_a += &product_a;
+
+            // B * r
+            let b_vec = b.get((i, j)).ok_or_else(|| {
+                PvwError::InvalidParameters(format!("CRS matrix access failed at ({}, {})", i, j))
+            })?;
+            let b_ij = Poly::from_coefficients(&b_vec, ctx).unwrap();
+            let product_b = &b_ij * &r_j_poly;
+            row_result_b += &product_b;
+        }
+
+        // Add noise: row_result_a + e1[i]
+        row_result_a += &e1_i_poly;
+        c1.push(row_result_a);
+
+        // Add noise: row_result_b + e2[i]
```

Comment:
we are adding the encoded msg and noise here (if we use the notation `x` you could write something like the following)

```suggestion
        // Add encoded message and noise: row_result_b + e2[i] + x[i]
```